pkg/kubelet/stats/host_stats_provider.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package stats
+
+import (
+	"fmt"
+	"path/filepath"
+
+	cadvisorapiv2 "github.com/google/cadvisor/info/v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	statsapi "k8s.io/kubelet/pkg/apis/stats/v1alpha1"
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+	"k8s.io/kubernetes/pkg/kubelet/kuberuntime"
+	"k8s.io/kubernetes/pkg/volume"
+)
+
+// PodEtcHostsPathFunc is a function to fetch a etc hosts path by pod uid.
+type PodEtcHostsPathFunc func(podUID types.UID) string
+
+// metricsProviderByPath maps a path to its metrics provider
+type metricsProviderByPath map[string]volume.MetricsProvider
+
+// HostStatsProvider defines an interface for providing host stats associated with pod.
+type HostStatsProvider interface {
```

Comment:
thanks for taking this from my earlier branch.

we can continue to evolve this, but its so much clearer to have this consolidated in a single module.